examples/technical_blogs/tartan_blog/build_and_run_networks.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python
+# Copyright (C) Codeplay Software Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License") with LLVM
+# Exceptions; you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://github.com/codeplaysoftware/oneapi-construction-kit/blob/main/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import os
+import sys
+import shutil
+import requests
+import argparse
+import subprocess
+
+parser = argparse.ArgumentParser(description='The purpose of this script is to build and run '
+                                 'VGG16 and Resnet50 networks with oneAPI and oneapi-construction-kit.')
+
+# Prereqs: portBLAS, SYCL-DNN, oneapi oneapi_construction_kit should be cloned and installed.
+parser.add_argument('--portBLAS_source', required=True, help='Path to portBLAS source')
+parser.add_argument('--SYCL_DNN_source', required=True, help='Path to SYCL-DNN source')
+parser.add_argument('--oneapi_construction_kit_build_dir', required=True, help='Path to oneAPI construction kit')
+parser.add_argument('--oneapi_base_toolkit', required=True, help='Path to oneAPI base toolkit')
+parser.add_argument('--input_image', required=True, help='Path to the input image to be processed using the VGG16 and Resnet50.')
+
+args = parser.parse_args()
+
+# Set environment variables
+os.environ["CMAKE_CXX_COMPILER"] = os.path.join(args.oneapi_base_toolkit, "compiler", "linux", "bin-llvm", "clang++")
+os.environ["CMAKE_C_COMPILER"] = os.path.join(args.oneapi_base_toolkit, "compiler", "linux", "bin-llvm", "clang")
+os.environ["LD_LIBRARY_PATH"] = f"{os.path.join(args.oneapi_base_toolkit, 'compiler', 'linux', 'lib')}:{os.path.join(args.oneapi_base_toolkit, 'compiler', 'linux', 'compiler', 'lib', 'intel64_lin')}:{os.environ.get('LD_LIBRARY_PATH', '')}"
+os.environ["OCL_ICD_FILENAMES"] = f"{os.path.join(args.oneapi_construction_kit_build_dir, 'lib', 'libCL.so')}"
+
+
+def run_cmake_and_ninja(cmake_command, build_dir):
+    try:
+        subprocess.run(cmake_command, check=True)
+        subprocess.run(["ninja", "-C", build_dir], check=True)
+    except subprocess.CalledProcessError as e:
+        print(f"Error executing command: {e}", file=sys.stderr)
```

Comment:
will this show the build output if it fails?

Reply:
Do we need to handle the error and call `sys.exit` here? If we don't catch it, if we let the error go up to `main`, it will be handled there and exit the process the same way, but it also allows callers to possibly handle the error.